conans/client/tools/scm.py

```diff
@@ -146,4 +171,99 @@ def get_branch(self):
             branch = status.splitlines()[0].split("...")[0].strip("#").strip()
             return branch
         except Exception as e:
-            raise ConanException("Unable to get git branch from %s\n%s" % (self.folder, str(e)))
+            raise ConanException("Unable to get git branch from %s: %s" % (self.folder, str(e)))
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache --non-interactive".format(command))
+
+    def clone(self, url, submodule=None):
+        assert submodule is None, "Argument not handled"
+        assert os.path.exists(self.folder), "It guaranteed to exists according to SCMBase::__init__"
+        params = " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
+
+        output = ""
+        if not os.path.exists(os.path.join(self.folder, '.svn')):
+            url = self.get_url_with_credentials(url)
+            command = 'co "{url}" . {params}'.format(url=url, params=params)
+            output += self.run(command)
+        output += self.run("revert . --recursive {params}".format(params=params))
```

Comment:
Isn't an additional "svn update" required after "svn revert"? I'm not sure, but I think one will only revert to the current local revision instead of remote HEAD. 
Another point. If I got it right this will always checkout and revert. But revert after fresh and clean checkout is not required. So the "revert" should be part of "else".

Reply:
About the `svn update`: there is always a `checkout` after the `clone` (see [here](https://github.com/conan-io/conan/blob/c35172b6ce865dcb60cd2553de2697db519f40fb/conans/client/source.py#L192)). As the SCM was first implemented having Git in mind the workflow is closer to it

About the `revert` after the `co`: here it is not a clean checkout because the folder can be already there (the `if` clause checks for the `.svn` directory). I would need to test it, but I'm pretty sure that a checkout over an existing folder will be dirty (conflicts and so on) so the `revert` is needed.

Reply:
IMHO git clone and checkout should be combined as one will always clone and checkout in a recipe.

About revert: you are checking for .svn (btw. calling svn info or something like that and catching the possible error would be more safe). If exists you only have to revert and update (checkout again will also perform an update). If not present you can be sure that everything is clean and you can just checkout. So change the if clause to "exists" then revert and finally perform a checkout.